### ISSUE TITLE
:books: [Documentation]: improve tagging criteria

### ISSUE THREAD

**Issue by mishaschwartz** [bird-house/birdhouse-deploy#476]
## Description

The tagging criteria when creating a new release mostly refers to how components interact, not how they impact the end-user:

https://github.com/bird-house/birdhouse-deploy/blob/1528ba14db5c476fca11033a97f7df99b74b0fc4/birdhouse/README.rst?plain=1#L324-L336

This is fine but a bit ambiguous we need to also consider how these changes affect users. I suggest changing the language to:

- **MAJOR**: (no change needed)
- **MINOR**: requires changes to configuration settings or when a user's workflow would be affected as a result of the change. Some examples:
    - an update to a component that requires changes to be made to the configuration settings for that same component
    - an update to a component that requires changes to be made to the configuration settings for other components
    - an update to any default setting for an existing component
    - an update to a Jupyterlab image, weaver deployment, etc. that requires changes to users' workflow files in order for them to run properly
- **PATCH**: requires no changes to configuration settings or introduces optional additional configurations. Some examples:
    - a new optional component is added
    - a fully backwards compatible change is made to an existing component

## References

| Information           | Value
| --------------------- | --------------------------------------------------------------
| Server/Platform URL   | all
| Related issues/PR     | #475
| Related documentation | https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/README.rst#tagging-policy

**Comment by tlvu** [comment]
I agree with the suggestion.  Also we should review any configs that is not generic (like Ouranos Jupyter specific image in https://github.com/bird-house/birdhouse-deploy/pull/475#issuecomment-2483949136) and if we can not make them optional, not enabled by default, we document them clearly they are not generic.

The components to review are those that were previously enable by default and now they are not.  The default values of their configs might be able to change to something more generic.
